### PR TITLE
chore(cli): add linter to detect potential spurious usage of owner user in cli tests

### PR DESCRIPTION
**Pull request by johnstcn**
Detects the following pattern where the CLI is initialized with a client authenticated as the "first user":

```
client := coderdtest.New(t, ...)
[...]
user := coderdtest.CreateFirstUser(t, client)
[...]
clitest.SetupConfig(t, client, root)
```

When updating the unit tests, I tried to stick to the principle of least privilege as much as possible. Note that this doesn't stop you from using the owner if needed, you just need to add a `//nolint:gocritic` with reasoning.

**Note:** In the course of going through this, I also noted an inaccuracy in our admin RBAC docs regarding updating other workspaces. Updated.

### CODE REVIEW

**Review by Emyrk — APPROVED** [review]
Diff doesn't seem as bad as I would have guessed. Sweet!

Curious how robust the rules.go is. I guess we never nest those 2 things in different scopes, so it should work pretty well.

**Review by mafredri — APPROVED** [review]
This solution turned out nice, awesome job!

File: cli/configssh_test.go [hunk 1]
```diff
@@ -75,9 +75,10 @@ func TestConfigSSH(t *testing.T) {
 			},
 		},
 	})
-	user := coderdtest.CreateFirstUser(t, client)
+	owner := coderdtest.CreateFirstUser(t, client)
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/configssh_test.go:78]: Updated naming to `owner` to be consistent across the codebase.

**Comment by johnstcn** [comment]
> Curious how robust the rules.go is. I guess we never nest those 2 things in different scopes, so it should work pretty well.

There does seem to be room for improvement, it didn't actually catch the port-forwarding thing because the calls to `coderdtest.CreateFirstUser` and `clitest.SetupConfig` were in different functions. I'll see about addressing that in a follow-up.

**Pull request merged**